timesketch/lib/analyzers/geoip.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2021 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Sketch analyzer plugin for geolocating IP addresses."""
+from __future__ import unicode_literals
+
+import ipaddress
+import logging
+
+from collections import defaultdict
+
+from flask import current_app
+
+import geoip2.database
+import geoip2.errors
+
+from timesketch.lib import emojis
+from timesketch.lib.analyzers import interface
+from timesketch.lib.analyzers import manager
+
+
+logger = logging.getLogger('timesketch.analyzers.geoip')
+
+
+class GeoIPSketchPlugin(interface.BaseAnalyzer):
+    """Sketch analyzer for geolocating IP addresses."""
+
+    NAME = 'geo_ip'
+    DISPLAY_NAME = 'Geolocate IP addresses'
+    DESCRIPTION = ('Find the approximate geolocation of an IP address using ' +
+                   'MaxMind GeoLite2')
+    DEPENDENCIES = frozenset(['feature_extraction'])
+    IP_FIELDS = ['ip_address', 'client_ip', 'host_ip']
```

Comment:
Please include the following fields: 
['ip', 'src_ip', 'dst_ip', 'source_ip', 'dest_ip', 'ip_address', 'client_ip', 'address', 'saddr', 'daddr', 'requestMetadata_callerIp', 'a_answer']

Reply:
Thanks, have now included additional fields.